aws-iot-thingtype/src/main/java/software/amazon/iot/thingtype/Translator.java

```diff
@@ -184,15 +185,44 @@ static List<ResourceModel> translateFromListResponse(final ListThingTypesRespons
                 .collect(Collectors.toList());
     }
 
+    static Map<String, String> translateTagstoMap(final Set<software.amazon.iot.thingtype.Tag> tags) {
+        if (tags == null) {
+            return new HashMap<>();
+        }
+
+        return tags.stream()
+                .collect(Collectors.toMap(
+                        software.amazon.iot.thingtype.Tag::getKey,
+                        software.amazon.iot.thingtype.Tag::getValue,
+                        (existing, replacement) -> existing,  // In case of duplicate keys, keep existing value
+                        HashMap::new
+                ));
+    }
+
     static Set<Tag> translateTagsToSdk(final Map<String, String> tags) {
-        if (tags == null) return Collections.emptySet();
-        return Optional.of(tags.entrySet()).orElse(Collections.emptySet())
+        if (tags == null || tags.isEmpty()) {
+            return Collections.emptySet();
+        }
+        return tags.entrySet()
                 .stream()
                 .map(tag -> Tag.builder().key(tag.getKey()).value(tag.getValue()).build())
                 .collect(Collectors.toSet());
     }
 
-    private static <T> Stream<T> streamOfOrEmpty(final Collection<T> collection) {
+    static Set<Tag> translateTagsToSdk(final Set<software.amazon.iot.thingtype.Tag> tags) {
+        if (tags == null || tags.isEmpty()) {
+            return Collections.emptySet();
+        }
+
+        return tags.stream()
+                .map(tag -> Tag.builder()
+                        .key(tag.getKey())
+                        .value(tag.getValue())
+                        .build())
+                .collect(Collectors.toSet());
+    }
+
```

Comment:
are these two methods the same?

Reply:
the are overloaded to take different argument types

Reply:
hmm, i see.. feels like we can avoid having repetitions